src/TestLogger/Extensions/XunitTestAdapter.cs

```diff
@@ -17,7 +17,7 @@ public class XunitTestAdapter : ITestAdapter
 
             // Process all the messages collected during the test run
             // If one ends with [SKIP], then the next message contains the skip reason.
-            var skippedTestNamesWithReason = new Dictionary<string, string>();
+            var skippedTestNamesWithReason = new Queue<(string testName, string skipReason)>();
```

Comment:
Was this change required because of tests running in parallel?

Reply:
Yes. This custom framework considers a test class as a test case whose methods are steps and always run sequentially, but the test cases usually run in parallel.